test/test_common/simulated_time_system_test.cc

```diff
@@ -289,60 +289,57 @@ TEST_P(SimulatedTimeSystemTest, WaitFor) {
   EXPECT_EQ(start_system_time_, time_system_.systemTime());
 
   // Run an event loop in the background to activate timers.
-  std::atomic<bool> done(false);
-  auto thread = Thread::threadFactoryForTest().createThread([this, &done]() {
-    while (!done) {
+  absl::Mutex mutex;
+  bool done(false);
+  auto thread = Thread::threadFactoryForTest().createThread([this, &mutex, &done]() {
+    for (;;) {
+      {
+        absl::MutexLock lock(&mutex);
```

Comment:
taste test, as this syntax works now (for you golang fans):
```
    for (;;) {
      if (absl::MutexLock lock(&mutex); done) {
        return;
      }
      base_scheduler_.run(Dispatcher::RunType::Block);
    }
```
Looking at this code is the first time it occurred to me to use it in C++.

Reply:
Oh yeah that's good. I will fix that in a follow up. I want to get this merged so we can see how we are doing with flakes.